src/desktop/apps/authentication/helpers.ts

```diff
@@ -112,19 +119,55 @@ export const setCookies = options => {
   }
 }
 
-export const getRedirect = type => {
+export async function apiAuthWithRedirectUrl(
+  response: any,
+  redirectPath: URL
+): Promise<URL> {
+  const redirectUrl = sd.APP_URL + redirectPath.pathname
+  const accessToken = (response.user || {}).accessToken
+  const appRedirectURL = new URL(redirectUrl)
+
+  // There isn't an access token when we don't have a valid session, for example,
+  // when the user is resetting their password.
+  if (!accessToken) return appRedirectURL
+
+  try {
+    const tokenResponse = await fetch(sd.API_URL + "/api/v1/me/trust_token", {
+      method: "POST",
+      headers: { "X-Access-Token": accessToken },
+    })
+
+    if (tokenResponse.ok) {
+      const responseBody = await tokenResponse.json()
+      const trustToken = responseBody["trust_token"]
```

Comment:
#minor. This can probably be `responseBody.trust_token` if more consistent.

Reply:
Is `responseBody.trust_token` more idiomatic JS/TS? In https://github.com/artsy/force/pull/4379/commits/30bf40e04408dbd4dfba5cfe8ae1f4d29cbc46b6 I refactored to use the `Response` type, which lead to `response["user"]` line, so I believe the code is more structurally consistent ATM.

Reply:
I tend to use the dot notation because it's more concise and readable (unless the key is a variable). ESLint has a [rule](https://eslint.org/docs/rules/dot-notation) recommending that as a best practice, but it seems like Force is not configured to use that yet.